### PR TITLE
Remove participant-side command deduplication [DPP-848]

### PR DESCRIPTION
**Pull request by tudor-da**
This PR removes:
* The `IndexSubmissionService`, its callers and implementation.
* The `DeduplicationStorageBackend`, its callers and implementation.
* The deduplication logic from `ApiSubmissionService` and `ReadOnlySqlLedger`
* The `participant_command_submissions` (directly for H2 and Oracle and via a migration for PG)

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V121__drop_participant_side_command_deduplication.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+-- Participant-side deduplication not supported anymore
```
Comment (rautenrieth-da) [inline ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V121__drop_participant_side_command_deduplication.sql:1]: Copyright header missing

File: ledger/participant-integration-api/src/main/resources/db/migration/oracle-appendonly/V1__Append_only_schema.sha256 [hunk 2]
```diff
@@ -1 +1 @@
-fe940b22140c0a1da0a3e54c0087c8872828e4cb08bdcc30bbaa1db84f98b62b
+93dd1c0237cb7f8b193aef03c88d3fd22e85c5e8d2c26adeff4c6eeeba0d500c
```
Comment (rautenrieth-da) [inline ledger/participant-integration-api/src/main/resources/db/migration/oracle-appendonly/V1__Append_only_schema.sha256:1]: Oracle already has [10 migrations](https://github.com/digital-asset/daml/tree/main/ledger/participant-integration-api/src/main/resources/db/migration/oracle-appendonly), why are you changing the first one? Shouldn't you add a new migration, just like Postgres?

**Review by rautenrieth-da — CHANGES_REQUESTED** [review]
Nice work removing all the dead code 🎉 

As mentioned in a comment, I think the Oracle schema change should be done in a new migration.

File: ledger/participant-integration-api/src/main/resources/db/migration/oracle-appendonly/V1__Append_only_schema.sha256 (hunk 2 above)
Comment (tudor-da) [reply ledger/participant-integration-api/src/main/resources/db/migration/oracle-appendonly/V1__Append_only_schema.sha256:1]: Oh, how did I miss that.

Comment (tudor-da) [reply ledger/participant-integration-api/src/main/resources/db/migration/oracle-appendonly/V1__Append_only_schema.sha256]: Thanks for catching this. Amended

File: ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V121__drop_participant_side_command_deduplication.sql (hunk 1 above)
Comment (tudor-da) [reply ledger/participant-integration-api/src/main/resources/db/migration/postgres-appendonly/V121__drop_participant_side_command_deduplication.sql:4]: Added

**Review by nicu-da — APPROVED** [review]
This brings a lot of joy 🎉 

One point about the feature descriptors we have. If participant deduplication is no longer a thing, can we change the `deduplication_type` feature flag to also remove the `SYNC_ONLY` option?

**Review by rautenrieth-da — APPROVED** [review]

**Comment by tudor-da** [comment]
> This brings a lot of joy 🎉
> 
> One point about the feature descriptors we have. If participant deduplication is no longer a thing, can we change the `deduplication_type` feature flag to also remove the `SYNC_ONLY` option?

Good point. Does the `ASYNC_AND_CONCURRENT_SYNC` still makes sense, then?

**Comment by nicu-da** [comment]
> `ASYNC_AND_CONCURRENT_SYNC`

Yes as that's canton specific

**Pull request merged**